src/NuGet.Clients/PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs

```diff
@@ -272,9 +272,14 @@ private void PerformPackageUpdatesOrReinstalls()
             foreach (var project in Projects)
             {
                 var installedPackages = await project.GetInstalledPackagesAsync(Token);
-                if (installedPackages.Select(installedPackage => installedPackage.PackageIdentity.Id)
-                                     .Any(installedPackageId => installedPackageId.Equals(packageId, StringComparison.OrdinalIgnoreCase)))
+
+                var installedPackage = installedPackages.Where(
```

Comment:
You can just get the package the first time to avoid enumerating this query twice, both when .Any() is called and also when .First() is called

``var installedPackage = installedPackages.FirstOrDefault(``